### PR TITLE
[reland][CI] switch torchbench to a pinned version

### PR DESCRIPTION
**Pull request by desertfire**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #96782

Summary: This is reland of https://github.com/pytorch/pytorch/pull/96553

cc @soumith @voznesenskym @penguinwu @anijain2305 @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx

### CODE REVIEW

File: benchmarks/dynamo/check_hf_bert_perf_csv.py [hunk 1]
```diff
@@ -17,7 +17,8 @@ def check_hf_bert_perf_csv(filename):
         model_name = row["name"]
         speedup = row["speedup"]
         # Reduce from 1.165 to 1.160, see https://github.com/pytorch/pytorch/issues/96530
-        if speedup < 1.160:
+        # Reduce from 1.160 to 1.145 after a transformer version upgrade, see https://github.com/pytorch/benchmark/pull/1406
+        if speedup < 1.145:
```
Comment (weiwangmeta) [inline benchmarks/dynamo/check_hf_bert_perf_csv.py:21]: During re-land, let's change it to 1.140, e.g. this action generated 1.1405 https://github.com/pytorch/pytorch/actions/runs/4420433748/jobs/7750698385

File: .ci/pytorch/test.sh [hunk 2]
```diff
@@ -320,12 +320,7 @@ test_single_dynamo_benchmark() {
       --output "$TEST_REPORTS_DIR/${name}_${suite}.csv"
     python benchmarks/dynamo/check_csv.py \
       -f "$TEST_REPORTS_DIR/${name}_${suite}.csv"
-    if [[ "${TEST_CONFIG}" != *cpu_accuracy* ]] && [[ "${TEST_CONFIG}" != *dynamic* ]]; then
-      # because I haven't tracked the cpu-side or dynamic expected artifacts yet, and need to differentiate filenames
-      python benchmarks/dynamo/check_graph_breaks.py \
-        --actual "$TEST_REPORTS_DIR/${name}_$suite.csv" \
-        --expected "benchmarks/dynamo/ci_expected_accuracy/${name}_${suite}${shard_id}.csv"
-    fi
+    # will add graph_break check back after collecting new graph break csv
```
Comment (huydhn) [inline .ci/pytorch/test.sh:323]: Will did a forward fix here https://github.com/pytorch/pytorch/pull/96780, so the conflict would need to be resolved

**Review by huydhn — APPROVED** [review]
Pre-stamp approval

**Pull request closed without merging**